### DISCUSSION TITLE
Auto-reload with Rhino

### DISCUSSION THREAD

**Discussion by mshunj** [Appsilon/rhino#502] [Q&A] [answered]
Are there guidelines on how to enable auto-reload with Rhino? Currently I can set options(shiny.autoreload =TRUE), change my code, and then on save of app.R, the ui gets updated. This is not ideal, as it should reload on save of any .R module. I've attempted options(shiny.autoreload =TRUE, shiny.autoreload.pattern = glob2rx("./app/**/*.R")) but that seems to break autoreload alltogether. Any guidance here would be appreciated.

**Comment by jakubnowicki** [comment, marked as answer]
Hi @mshunj . The main challenge in auto-reloading in Rhino is caused by `box`, as it uses cache for modules. Rhino purges cache on application start, but this does not happen if you only reload the application without restarting it. There is already a [draft solution](https://github.com/klmr/box/pull/246) in `box` that should make the auto-reload possible. Once this is incorporated in the `box` release, we will start working on enabling it in Rhino.
